geoh5py/ui_json/input_file.py

```diff
@@ -361,23 +369,28 @@ def geoh5(self):
         return self._geoh5
 
     @geoh5.setter
-    def geoh5(self, geoh5: Workspace | None):
-        if geoh5 is not None:
-            if self._geoh5 is not None:
-                raise UserWarning(
-                    "Attribute 'geoh5' already set. "
-                    "Consider creating a new InputFile from arguments."
-                )
-
-            if not isinstance(geoh5, Workspace):
-                raise ValueError(
-                    "Input 'geoh5' must be a valid :obj:`geoh5py.workspace.Workspace`."
-                )
+    def geoh5(self, geoh5: Workspace | str | Path | None):
+        if geoh5 is None:
+            return
+
+        if self._geoh5 is not None:
+            raise UserWarning(
+                "Attribute 'geoh5' already set. "
+                "Consider creating a new InputFile from arguments."
+            )
 
-        self._geoh5 = geoh5
+        if isinstance(geoh5, (str, Path)):
+            self._geoh5 = Workspace(geoh5, mode="r")
+        elif isinstance(geoh5, Workspace):
+            self._geoh5 = geoh5
```

Comment:
@benk-mira , @domfournier , in combination with `plain_data()` for param-sweeps, this method now allows for str or Path, and instantiate the workspace from it